keyboards/h0oni/deskpad/keymaps/default/keymap.c

```diff
@@ -0,0 +1,223 @@
+    [TD_MPRV_LEFT] = ACTION_TAP_DANCE_DOUBLE(KC_MPRV, KC_LEFT),
+    [TD_SEARCH_REFRESH] = ACTION_TAP_DANCE_DOUBLE(KC_WREF, KC_WSCH),
+    [QUAD_LAYER_SWITCH] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, layer_finished, layer_reset),
+    [QUAD_CVXA] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, cvxa_finished, cvxa_reset)
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case YOUTUBE:
+            if (record->event.pressed) {
+                tap_code16(C(KC_L));
+                SEND_STRING("https://www.youtube.com/");
+            } else {
+                tap_code(KC_ENT);
+            }
+            break;
+        case FACEBOOK:
+            if (record->event.pressed) {
+                tap_code16(C(KC_L));
+                SEND_STRING("https://www.facebook.com/");
+            } else {
+                tap_code(KC_ENT);
+            }
+            break;
+        case VALORANT:
+            if (record->event.pressed) {
+                tap_code16(G(KC_S));
+                SEND_STRING("valorant");
+            } else {
+                tap_code(KC_ENT);
+            }
+            break;
+        case DISCORD:
+            if (record->event.pressed) {
+                tap_code16(G(KC_S));
+                SEND_STRING("discord");
+            } else {
+                tap_code(KC_ENT);
+            }
+            break;
+        case VSCODE:
+            if (record->event.pressed) {
+                tap_code16(G(KC_S));
+                SEND_STRING("vscode");
+            } else {
+                tap_code(KC_ENT);
+            }
+            break;
+    }
+    return true;
+}; 
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_all(
+      TD(TD_CUT_REDO), TD(TD_MPRV_LEFT), TD(TD_PLAY_PAUSE_MUTE), TD(TD_MNXT_RIGHT), TD(QUAD_CVXA), TD(QUAD_LAYER_SWITCH)
+      ),
+
+  [1] = LAYOUT_all(
+      YOUTUBE, KC_WBAK, TD(TD_SEARCH_REFRESH), KC_WFWD, FACEBOOK, TD(QUAD_LAYER_SWITCH)
+      ),
+
+  [2] = LAYOUT_all(
+      A(KC_F4), SGUI(KC_S), KC_MYCM, LCA(KC_DEL), KC_CALC, TD(QUAD_LAYER_SWITCH)
+      ),
+  
+  [3] = LAYOUT_all(
+      C(KC_SLSH), VALORANT, VSCODE, DISCORD, LSA(KC_A), TD(QUAD_LAYER_SWITCH)
+      ),
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    const layer_state_t curr_layer = get_highest_layer(layer_state);
+    if(curr_layer == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    else if(curr_layer == 1) {
+        if (clockwise) {
+            tap_code(KC_WH_D);
+        } else {
+            tap_code(KC_WH_U);
+        }
+    }
+    else if(curr_layer == 1) {
+        if (clockwise) {
+            tap_code16(C(KC_WH_U));
+        } else {
+            tap_code16(C(KC_WH_D));
+        }
+    }
+    else {
+        if (clockwise) {
+            tap_code(KC_BRIU);
+        } else {
+            tap_code(KC_BRID);
+        }
+    }
+    return false;
+}
```

Comment:
this should be moved to the keyboards c file, and switched to the kb level function
```suggetion
```